test/blackbox-tests/test-cases/corrections/run.t

```diff
@@ -0,0 +1,58 @@
+  $ echo '(lang dune 2.0)' > dune-project
+  $ echo file-contents > text-file
+
+
+It's OK if there's no correction:
+
+  $ cat > dune <<EOF
+  > (alias (name no_correction)
+  >   (deps )
+  >   (action (progn (diff? text-file text-file-corrected)))
+  > )
+  > EOF
+  $ dune build @no_correction
+
+# CR-someday aalekseyev:
+# Weird: if dependency is missing, dune fails to report a correction.
+# This is an idiom used in the wild,
+# and in fact the docs suggest that one should do something like this,
+# and you can't even specify optional targets at all.
+
+  $ cat > dune <<EOF
+  > (alias (name correction1)
+  >   (deps)
+  >   (action
+  >     (progn (bash "> text-file-corrected echo corrected-contents-1")
```

Comment:
You can use `(write-file ..)` here and no bash will be necessary. Then everything will work on all exotic platforms like OSX, BSD, WASM, Win 95, etc :)

Reply:
Sadly, that doesn't work. Dune then complains that aliases can't have dependencies.

Reply:
(I'm sure bash could be run on most of those as well :-) )

Reply:
Ah well in that case can we try `(system ..)`. That should use `sh` which I think is a bit more portable than bash and can be safely assumed to be identical on more platforms.